### PR TITLE
Update Third Party Javascript Management Cheat Sheet

### PR DESCRIPTION
**Pull request by jimweiler1775**
Added new text at introduction defining new attack surface metrics to demonstrate risk from third party javascript.

### CODE REVIEW

File: cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
 # Introduction
 
+The attack surface of a web application is generally considered to be the size of the web site, aka the number of page urls, forms, including the number of parameters and parameter values for single page apps, as well as the amount of HTML generated by the server. Various definitions have also included the protocol between the client and the server (TLS). These are basically server side measures. A different measure includes the business logic complexity and the number of use cases or user stories on the site. A use case will involve a sequence of pages or URL parameter values, as well as the metadata of the user to initialize the use case; referrer, user agent, authorization or loyalty member level, language, geolocation etc.
+
+I define the use case attack surface as the combination of three values:
```
Comment (righettod) [inline cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md]: Use impersonal mode,

File: cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+UCAS1:the number of bytes of code executed for a given use case, user action or user story; (more code provides more risk of insecure code).
```
Comment (righettod) [inline cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md]: Can you use bullet point for the different **UCAS** in order to made it more easy to read ?

Comment (jimweiler1775) [reply cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md]: I agree, it felt strange to say 'I define',  I'll change it  and add bullets

Comment (jimweiler1775) [reply cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md]: Also, I have never used Git, so I'll have to do some reading to figure out how to change existing text.

Comment (jimweiler1775) [reply cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md]: I made the changes on the patch1 branch, I think.  Do I open a new pull request or do you approve the edited branch?

Comment (righettod) [reply cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md]: You have the right thing, pushing to your fork update this PR so it's OK. 👍

File: cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+
+The use case attack surface can be defined as the combination of three values:
+
+1. UCAS1:the number of bytes of code executed for a given use case, user action or user story; (more code provides more risk of insecure code).
+
+2. UCAS2:the number of source code files involved in executing that use case; (more source files implies more developers and developer teams with different levels of secure code knowledge, especially with microservice architectures).
```
[the remark this answers is not in the document]
Comment (jimweiler1775) [reply cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md]: Client side pages do have lots of smaller files from many different sources, that is exactly the point. This has nothing to do with npn modules or any server side files. On 3 different booking path pages, Marriott has 100 js files from 34 vendors, 103 js files from 46 vendors, 157 js files from 47 vendors. Try looking at Chrome dev tools sometime on an ecommerce page. I'm happy to delete the TLS reference.

File: cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md [hunk 4]
```diff
@@ -1,5 +1,23 @@
 # Introduction
 
+The attack surface of a web application is generally considered to be the size of the web site, aka the number of page urls, forms, including the number of parameters and parameter values for single page apps, as well as the amount of HTML generated by the server. Various definitions have also included the protocol between the client and the server (TLS). These are basically server side measures. A different measure includes the business logic complexity and the number of use cases or user stories on the site. A use case will involve a sequence of pages or URL parameter values, as well as the metadata of the user to initialize the use case; referrer, user agent, authorization or loyalty member level, language, geolocation etc.
+
+The use case attack surface can be defined as the combination of three values:
```
[the remark this answers is not in the document]
Comment (jimweiler1775) [reply cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md]: as I said in the text, I'm suggesting a  new metric.

File: cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md [hunk 5]
```diff
@@ -1,5 +1,23 @@
...
+
+1. UCAS1:the number of bytes of code executed for a given use case, user action or user story; (more code provides more risk of insecure code).
+
+2. UCAS2:the number of source code files involved in executing that use case; (more source files implies more developers and developer teams with different levels of secure code knowledge, especially with microservice architectures).
+
+3. UCAS3: the number of different vendors that have code executing for that use case. (the more open source, marketing and analytic vendor code and proprietary 3rd party libraries that are used, the greater the insecure code risk).
```
[the remark this answers is not in the document]
Comment (jimweiler1775) [reply cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.md]: yeah, more consistency is better - a hyphen and no brackets is good

**Comment by mackowski** [comment]
@jimweiler1775 Could you update your PR or express your thoughts about suggested changes please?

**Comment by ThunderSon** [comment]
Following Author's unavailability, and in the current state of the PR, the team can't see this PR forward, thus it will be closed. For further discussions, an issue should be opened.

**Pull request closed without merging**